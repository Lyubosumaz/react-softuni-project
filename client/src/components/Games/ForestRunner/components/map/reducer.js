import { generalActions } from './actions';

const initialState = {
    tiles: [],
};

const mapReducer = (state = initialState, action) => {
    switch (action.type) {
        case generalActions.SET_TILES:
            return {
                ...action.payload,
            };
        default:
            return state;
    }
};

export default mapReducer;
