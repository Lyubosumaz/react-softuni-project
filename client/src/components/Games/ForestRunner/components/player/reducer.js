const initialState = {
    position: [0, 0],
    spriteLocation: '0px 0px',
    default: 'EAST',
    walkIndex: 0,
};

const playerReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'MOVE_PLAYER':
            return {
                ...state,
                ...action.payload,
            }
        case 'RESET_PLAYER_LOCATION':
            return {
                ...state,
                ...initialState
            }
        default:
            return state;
    };
};

export default playerReducer;