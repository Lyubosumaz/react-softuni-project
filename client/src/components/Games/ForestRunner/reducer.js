const initialState = {
    inGame: false,
    gold: 0,
    item: [],
    time: 0,
    handleGameTimer: false,
    level: 1,
    gameItems: [],
    inventorySellItem: false,
    inventoryEquipItem: false,
    characterRemoveItem: false,
};

const gameReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'SAVE_GAME_ITEMS_LOOT':
            return {
                ...state,
                gameItems: action.payload,
            };
        case 'GAME_TIMER_STATE':
            return {
                ...state,
                handleGameTimer: action.payload,
            };
        case 'SET_TIME':
            return {
                ...state,
                time: action.payload,
            };
        case 'OPEN_GOLD_CHEST':
            return {
                ...state,
                gold: (state.gold += action.payload),
            };
        case 'OPEN_ITEM_CHEST':
            return {
                ...state,
                item: state.item.concat(action.payload),
            };
        case 'START_THE_GAME':
            return {
                ...state,
                inGame: action.payload,
            };
        case 'END_THE_GAME':
            return {
                ...state,
                inGame: action.payload,
                level: 2,
            };
        case 'RESET_GAME_LEVEL':
            return {
                ...state,
                gold: 0,
                item: [],
                time: 0,
            };
        case 'CHARACTER_SELL_ITEM':
            return {
                ...state,
                inventorySellItem: action.payload,
            };
        case 'CHARACTER_EQUIP_ITEM':
            return {
                ...state,
                inventoryEquipItem: action.payload,
            };
        case 'CHARACTER_REMOVE_ITEM':
            return {
                ...state,
                characterRemoveItem: action.payload,
            };
        default:
            return state;
    }
};

export default gameReducer;
