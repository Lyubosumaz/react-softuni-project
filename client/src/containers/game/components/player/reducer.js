const initialState = {
    inGame: false,
    position: [0, 0],
    spriteLocation: '0px 0px',
    default: 'EAST',
    walkIndex: 0,
    final: false,
    gold: 0,
    items: []
};

const playerReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'MOVE_PLAYER':
            return {
                ...state,
                ...action.payload,
            }
        case 'OPEN_CHEST':
            return {
                ...state,
                gold: state.gold += action.payload,
            };
        case 'FINAL':
            return;
        case 'START':
            return {
                ...state,
                inGame: action.payload,
            }
        default:
            return state;
    };
};

export default playerReducer;