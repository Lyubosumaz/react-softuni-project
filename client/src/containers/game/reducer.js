const initialState = {
    inGame: false,
    final: false,
    gold: 0,
    item: [],
    time: 0,
    level: 1,
    gameItems: [],
    inventorySellItem: false,
    inventoryEquipItem: false,
    characterRemoveItem: false
};

const gameReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'SAVE_GAME_ITEMS':
            return {
                ...state,
                gameItems: action.payload,
            }
        case 'START':
            return {
                ...state,
                inGame: action.payload,
            }
        case 'OPEN_GOLD_CHEST':
            return {
                ...state,
                gold: state.gold += action.payload,
            };
        case 'OPEN_ITEM_CHEST':
            return {
                ...state,
                item: state.item.concat(action.payload),
            };
        case 'GET_TIME':
            return {
                ...state,
                time: action.payload,
            };
        case 'FINAL':
            return {
                ...state,
                inGame: action.payload,
                level: 2,
            };
        case 'RESET_GAME':
            return {
                ...state,
                final: false,
                gold: 0,
                item: [],
                time: 0,
                // level: 1
            };
        case 'CHARACTER_SELL_ITEM':
            return {
                ...state,
                inventorySellItem: action.payload
            };
        case 'CHARACTER_EQUIP_ITEM':
            return {
                ...state,
                inventoryEquipItem: action.payload
            };
        case 'CHARACTER_REMOVE_ITEM':
            return {
                ...state,
                characterRemoveItem: action.payload
            };
        default:
            return state;
    };
};

export default gameReducer;