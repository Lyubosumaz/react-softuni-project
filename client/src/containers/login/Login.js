import React, { useState } from 'react';
import { useHistory } from "react-router-dom";
import { connect } from 'react-redux';

import userService from '../../services/user-services';
import schema from './login-validations';

function Login(props) {
    const username = useFormInput('');
    const password = useFormInput('');
    const [errors, setErrors] = useState({});
    const history = useHistory();

    return (
        <form>
            <div className="form-container">
                <h1>Login</h1>
                <p>Please enter your Username and Password.</p>

                <div className="form-div-container">
                    <label for="username"><b>Username:</b></label>
                    <input type="text" placeholder="Enter your Username" name="username" className="form-input" id="username" {...username} />
                    {errors.username && <div className="form-input-error">{errors.username[0]}</div>}
                </div>

                <div className="form-div-container">
                    <label for="password"><b>Password:</b></label>
                    <input type="password" placeholder="Enter your Password" name="password" className="form-input" id="password" {...password} />
                    {errors.password && <div className="form-input-error">{errors.password[0]}</div>}
                </div>

                <div>
                    <button type="submit" className="form-action-btn" onClick={handleSubmit}>Login</button>
                </div>

                <div className="form-info-container">
                    <p>Don't have an account? <a href="/register">Create account</a>.</p>
                </div>
            </div>
        </form>
    );

    function handleSubmit(e) {
        e.preventDefault();

        const data = {
            username: username.value,
            password: password.value,
        };
        const hasErrors = Object.keys(errors).filter(key => errors[key].length > 0);

        if (hasErrors.length === 0 && data.username && data.password) {
            userService.login(data)
                .then(() => {
                    props.setLoginValue();
                    history.push('/home');
                }).catch(err => {
                    console.log(err)
                    setErrors({ ...errors, password: [err] });
                })
        }
    };

    function useFormInput(initialValue) {
        const [value, setValue] = useState(initialValue);

        function handleChange(event) {
            setValue(event.target.value);
            validate(event);
        };

        function validate(event) {
            const name = event.target.id;

            schema.fields[name].validate(event.target.value, { abortEarly: false })
                .then(() => {
                    setErrors({ ...errors, [name]: [] });
                })
                .catch((err) => {
                    setErrors({ ...errors, [name]: err.errors });
                });
        };

        return { value, onChange: handleChange };
    };
};

function mapStateToProps(state) {
    return {
        isLogin: state.login.isLogin,
    };
}
function mapDispatchToProps(dispatch) {
    return {
        setLoginValue: () => dispatch({
            type: 'USER_LOGGED_IN'
        }),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);