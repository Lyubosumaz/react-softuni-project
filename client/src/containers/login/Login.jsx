import { useState } from 'react';
import { connect } from 'react-redux';
import { setLoginValue } from './actions';
import history from '../../utils/history';
import handleRoute from '../../utils/handleRoutes';
import http from '../../services/http';
import { toast } from 'react-toastify';
import schema from './login-validations';

function Login(props) {
    const username = useFormInput('');
    const password = useFormInput('');
    const [errors, setErrors] = useState({});

    function useFormInput(initialValue) {
        const [value, setValue] = useState(initialValue);

        function handleChange(event) {
            setValue(event.target.value);
            validate(event);
        }

        function validate(event) {
            const name = event.target.id;

            schema.fields[name]
                .validate(event.target.value, { abortEarly: false })
                .then(() => {
                    setErrors({ ...errors, [name]: [] });
                })
                .catch((err) => {
                    setErrors({ ...errors, [name]: err.errors });
                });
        }
        return { value, onChange: handleChange };
    }

    function handleSubmit(e) {
        e.preventDefault();
        const data = {
            username: username.value,
            password: password.value,
        };
        const hasErrors = Object.keys(errors).filter((key) => errors[key].length > 0);

        if (hasErrors.length === 0 && data.username && data.password) {
            http.User.login(data)
                .then((user) => {
                    props.setLoginValue(user);
                    toast('You have Logged successfully!', {
                        type: toast.TYPE.SUCCESS,
                    });
                    history.push('/home');
                })
                .catch((err) => {
                    setErrors({ ...errors, password: [err] });
                });
        }
    }

    return (
        <div className="login-container">
            <form>
                <h1>Login</h1>
                <p>Please enter your Username and Password.</p>

                <div className="form-div-container">
                    <label htmlFor="username">
                        <b>Username:</b>
                    </label>
                    <input type="text" placeholder="Enter your Username" name="username" className="form-input" id="username" {...username} />
                    {errors.username && <div className="form-input-error">{errors.username[0]}</div>}
                </div>

                <div className="form-div-container">
                    <label htmlFor="password">
                        <b>Password:</b>
                    </label>
                    <input type="password" placeholder="Enter your Password" name="password" className="form-input" id="password" {...password} />
                    {errors.password && <div className="form-input-error">{errors.password[0]}</div>}
                </div>

                <div>
                    <button type="submit" className="form-action-btn" onClick={handleSubmit}>
                        Login
                    </button>
                </div>

                <div className="info-container">
                    <p>
                        Don't have an account?{' '}
                        <button className="info-button" onClick={handleRoute('/register')}>
                            Create account
                        </button>
                        .
                    </p>
                </div>
            </form>
        </div>
    );
}

function mapStateToProps(state) {
    return {
        isLogin: state.user.isLogin,
    };
}

function mapDispatchToProps(dispatch) {
    return {
        setLoginValue: (data) => dispatch(setLoginValue(data)),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);
