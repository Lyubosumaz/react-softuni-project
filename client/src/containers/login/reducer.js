const initialState = {
    isLogin: false,
    userId: null,
}

const loginReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'USER_LOGGED_IN':
            return {
                ...state,
                isLogin: true,
                userId: action.payload._id
            }
        default:
            return state;
    }
}

export default loginReducer;