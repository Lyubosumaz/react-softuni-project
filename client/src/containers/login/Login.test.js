import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { mount } from 'enzyme';
import reducer from './reducer';
import Login from './Login';
import expect from 'expect';

describe('Login Component', () => {
    let getWrapper;
    beforeEach(() => {
        const mockStore = createStore(reducer, { user: { isLogin: true } });
        jest.mock('history', () => ({
            push: jest.fn(fn => fn()),
        }));
        getWrapper = () => mount(
            <Provider store={mockStore}>
                <Login />
            </Provider>
        );
    });

    it('Renter Login', () => {
        const wrapper = getWrapper();
        expect(wrapper).toBeTruthy();
    });

    it('Should render without errors main-container', () => {
        const wrapper = getWrapper().find('.main-container');
        expect(wrapper.length).toBe(1);
    });

    it('Checks for html changes', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    });
});
