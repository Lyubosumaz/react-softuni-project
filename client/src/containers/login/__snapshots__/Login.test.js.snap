// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Checks for html changes 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Login)>
    <Login
      isLogin={true}
      setLoginValue={[Function]}
    >
      <div
        className="main-container"
      >
        <form>
          <h1>
            Login
          </h1>
          <p>
            Please enter your Username and Password.
          </p>
          <div
            className="form-div-container"
          >
            <label
              htmlFor="username"
            >
              <b>
                Username:
              </b>
            </label>
            <input
              className="form-input"
              id="username"
              name="username"
              onChange={[Function]}
              placeholder="Enter your Username"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-div-container"
          >
            <label
              htmlFor="password"
            >
              <b>
                Password:
              </b>
            </label>
            <input
              className="form-input"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Enter your Password"
              type="password"
              value=""
            />
          </div>
          <div>
            <button
              className="form-action-btn"
              onClick={[Function]}
              type="submit"
            >
              Login
            </button>
          </div>
          <div
            className="info-container"
          >
            <p>
              Don't have an account? 
              <button
                className="info-button"
                onClick={[Function]}
              >
                Create account
              </button>
              .
            </p>
          </div>
        </form>
      </div>
    </Login>
  </Connect(Login)>
</Provider>
`;
